plugins/inputs/lotus/lotus.go

```diff
@@ -296,13 +335,38 @@ func recordTipsetStatePoints(ctx context.Context, api api.FullNode, acc telegraf
 	return nil
 }
 
+func recordMpoolUpdatePoints(ctx context.Context, acc telegraf.Accumulator, newMpoolUpdate api.MpoolUpdate, receivedAt time.Time, receivedFrom peer.ID) error {
+	var updateType string
+	switch newMpoolUpdate.Type {
+	case api.MpoolAdd:
+		updateType = "MpoolAdd"
+	case api.MpoolRemove:
+		updateType = "MpoolRemove"
+	default:
+		return fmt.Errorf("unknown mpool update type: %v", newMpoolUpdate.Type)
+	}
+
+	acc.AddFields("chain_mpool",
+		map[string]interface{}{
+			"message_size": newMpoolUpdate.Message.Size(),
+			"recorded_at":  receivedAt.UnixNano(),
+		},
+		map[string]string{
+			"message_cid_tag":       newMpoolUpdate.Message.Cid().String(),
+			"mpool_update_type_tag": updateType,
```

Comment:
This wasn't specified in the acceptance criteria: Happy to remove if unwanted. Its value indicates if the message is being added to the pool or removed from the pool.

Reply:
I was going to ask for it to be removed as I believe it's redundant to information we capture with chainwatch (since we can assume a msg is removed from the pool when it's added to a block), but I think we want the datasets to be independent in case we come across a sentinel configuration which does not have chainwatch included. Thanks for the consideration.

Reply:
Does that mean keep it or remove it?

It's also worth noting that messages can be removed (what lotus calls "pruning") from the pool for reasons other than being included in a block, e.g. invalid fields, pool being too full, insufficient gas, etc. I think keeping a record of this adds value.

Reply:
Sorry, yes keep it.